device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/MultiplexingClient.java

```diff
@@ -339,13 +481,69 @@ public void registerDeviceClients(Iterable<DeviceClient> deviceClients) throws I
      */
-    public void unregisterDeviceClient(DeviceClient deviceClient) throws InterruptedException
+    public void unregisterDeviceClient(DeviceClient deviceClient) throws InterruptedException, MultiplexingClientException
+    {
+        this.unregisterDeviceClient(deviceClient, DEFAULT_UNREGISTRATION_TIMEOUT_MILLISECONDS);
+    }
+
+    /**
+     * Unregister a device client from this multiplexing client. This method may be called before or after opening the
+     * multiplexed connection.
+     * <p>
+     * Users should use {@link #unregisterDeviceClients(Iterable)} for unregistering multiple devices as it has some
+     * performance improvements over repeatedly calling this method. This method blocks on each unregistration, whereas
+     * {@link #registerDeviceClients(Iterable)} blocks on all of the unregistrations after starting them all asynchronously.
+     * <p>
+     * If the multiplexed connection is already open, then this call will close the AMQP device session associated with
+     * this device, but it will not close any other registered device sessions or the multiplexing client itself.
+     * <p>
+     * If the multiplexed connection is already open, and this call would unregister the last device client,
+     * the multiplexed connection will remain open. The multiplexed connection can only be closed by calling
+     * {@link #close()}
+     * <p>
+     * Once a device client is unregistered, it may be re-registered to this or any other multiplexing client. It cannot
+     * be used in non-multiplexing scenarios or used by the deprecated {@link TransportClient}.
+     * <p>
+     * @param deviceClient The device client to unregister from this multiplexing client.
+     * @param timeoutMilliseconds How long (in milliseconds) to let this operation wait for all unregistrations to complete. If this threshold is passed, a {@link com.microsoft.azure.sdk.iot.device.exceptions.MultiplexingClientDeviceRegistrationTimeoutException} is thrown.
+     * @throws InterruptedException If the thread gets interrupted while waiting for the unregistration to succeed.
+     * @throws com.microsoft.azure.sdk.iot.device.exceptions.MultiplexingClientDeviceRegistrationTimeoutException If the unregistration takes longer than the provided timeout allows.
+     * @throws MultiplexingClientException If any other Exception is thrown, it will be nested into this exception.
+     */
+    public void unregisterDeviceClient(DeviceClient deviceClient, long timeoutMilliseconds) throws InterruptedException, MultiplexingClientException
     {
         Objects.requireNonNull(deviceClient);
         List<DeviceClient> clientList = new ArrayList<>();
         clientList.add(deviceClient);
-        unregisterDeviceClients(clientList);
+        unregisterDeviceClients(clientList, timeoutMilliseconds);
+    }
+
+    /**
+     * Unregister device clients from this multiplexing client. This method may be called before or after opening the
```

Comment:
nit: maybe add the word `multiple` at the beginning ...
`Unregister one or many device clients ...`